apps/system/js/screen_manager.js

```diff
@@ -333,6 +342,21 @@
       }
     },
 
+    turnShadeOn: function scm_turnShadeOn() {
+      this._screenShade = true;
+      this.setScreenBrightness(0, true);
+    },
+
+    turnShadeOff: function scm_turnShadeOff() {
+      // Bug 1185036: If the screen was disable with the shade on, we won't
+      // the screen won't turn on again until the user puts their finger on the
+      // screen again.
+      var brightness = parseFloat(this._savedBrightness) ||
+        parseFloat(this._userBrightness) || 1;
```

Comment:
Remove these `parseFloat()`. It's not possible these value can be anything other than numbers.